core/common/src/main/java/alluxio/network/protocol/databuffer/DataByteBuffer.java

```diff
@@ -52,6 +54,8 @@ public ByteBuffer getReadOnlyByteBuffer() {
 
   @Override
   public void release() {
-    // Nothing we need to release explicitly, let GC take care of all objects.
+    if (mBuffer.isDirect()) {
+      BufferUtils.cleanDirectBuffer(mBuffer);
```

Comment:
I don't think we have to leave a TODO for ref count, but maybe leave a comment about sharing/not sharing?